packages/manager/modules/core/src/index.js

```diff
@@ -158,8 +165,12 @@ export const registerCoreModule = (environment) => {
     .run(
       /* @ngInject */ ($rootScope, coreConfig) => {
         $rootScope.$on('lang.onChange', (event, { lang }) => {
-          coreConfig.setUserLocale(lang);
-          window.location.reload();
+          if (onLocaleChange) {
+            onLocaleChange(lang);
```

Comment:
don't you want to call
```js
coreConfig.setUserLocale(lang);
```
in both cases ?

Reply:
I think you have right (even it's working without >_<)

Reply:
In fact it's not necessary :-)

As `onLocaleChange` callback does the same as `coreConfig.setUserLocale`. It sets the user locale of `environment` object through i18n plugin : https://github.com/ovh/manager/pull/6098/files#diff-17953c7ed48e08a390075d0a4ab263796472f2d1e6c2a1d167a7eef2fb72b85aR54-R59